### PR TITLE
Add logger

### PR DESCRIPTION
**Pull request by huchenlei**
This PR adds a logger to help distinguish different levels of logs. The default log level is set to `INFO`. 

Example log of ControlNet version:
```
2023-06-02 18:01:51,483 - ControlNet - INFO - ControlNet v1.1.213
```

We should be able to slowly migrate to using the logger from unconditional print.

### CODE REVIEW

File: scripts/logging.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    logging.Formatter("%(asctime)s - %(name)s - %(levelname)s - %(message)s")
+)
+logger.addHandler(handler)
+
+
+if hasattr(shared.cmd_opts, "controlnet_loglevel"):
+    loglevel_string = shared.cmd_opts.controlnet_loglevel
+else:
+    print("Warning: `controlnet_loglevel` not available from A1111")
+    loglevel_string = "DEBUG"
```
Comment (ljleb) [inline scripts/logging.py:20]: Why not only use `getattr(shared.cmd_opts, "controlnet_loglevel", "INFO")`? AFAIK the if branch should only run when users do not do a full restart of the webui. Also, is there a reason we are falling back on debug here but using info by default?

**Comment by ljleb** [comment]
Looks good overall, although we should probably reload scripts/logging.py in scripts/controlnet.py
I have seen some efforts on the side of a1111 webui to use logging as well. However I am not sure where they are at with this and whether a part of the cake will be usable by extensions.

**Comment by huchenlei** [comment]
I think we should be fine as we are not directly using the root logger. I'd be happy if A1111 adds a global loglevel cmd argument that we can inherit from.

File: scripts/logging.py (hunk 1 above)
Comment (huchenlei) [reply scripts/logging.py:20]: Done.

**Pull request merged**